### DISCUSSION TITLE
Code-Switching Manifest

### DISCUSSION THREAD

**Discussion by SuperReem** [NVIDIA-NeMo/NeMo#10832] [Q&A]
Hi,

Is this manifest configuration for the text field correct for code-switching fine-tuning? Also does language model training with aggregate tokenizer support this?

```
{
  "audio_filepath": "seg_36_360-45_870.wav",
  "duration": "9.509999999999998",
  "text": [
    {"lang": "ar", "str": "و"},
    {"lang": "ar", "str": "ايه"},
    {"lang": "en", "str": "I"},
    {"lang": "en", "str": "think"},
    {"lang": "ar", "str": "انه"},
    {"lang": "ar", "str": "رح"},
    {"lang": "ar", "str": "يكون"},
    {"lang": "ar", "str": "مشهد"},
    ]
}
```

**Comment by SuperReem** [comment]
@titu1994 @nithinraok

**Comment by KunalDhawan** [comment]
Hi @SuperReem,
Yes, the manifest is in accordance with the multi-sample LID format for code-switching ([ref](https://github.com/NVIDIA/NeMo/tree/main/scripts/speech_recognition/code_switching)). 
You might also want to check out online code switching-dataset functionality, if it fits your use case - https://github.com/NVIDIA/NeMo/pull/6579
Sorry, we currently don't support LM training with code-switched text + aggregate tokenizer.
